dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProjectMapper.xml

```diff
@@ -169,6 +169,13 @@
         <if test="userId != 0">
             right join t_ds_relation_project_user rpu on p.id = rpu.project_id
             where rpu.user_id = #{userId}
+            union
+            select
+            <include refid="baseSqlV2">
+                <property name="alias" value="p2"/>
+            </include>
+            from t_ds_project p2
+            where p2.user_id = #{userId}
```

Comment:
I think we should refactor the api used by the dependent node instead of using union all. cc @caishunfeng @zhongjiajie @EricGao888 @ruanwenjun

Reply:
It's not recommend to use union, I hope we can do all the union, join operation in memory, we need to make the sql easier.

Reply:
@ruanwenjun @SbloodyS If you don't use union, it may not be easy to get pagination in the future. I'm thinking if the creator is also put into t_ds_relation_project_user and distinguishes whether to create a user or a normal user

Reply:
I prefer to doing this in service level instead of with sql. If we have complex logic in both service and sql level, it is much easier to introduce new bugs.

Reply:
I found that `queryAllProject` is also used by the working group. And it should be permission controlled.

In my opinion, The project list in dependent node's permissions should not be restricted. I've created a new issue to describe it. We would like to release it in the next month's 3.1.0 version. Would you like to take it? If not, I'll take it by myself. @limaiwang

Reply:
@SbloodyS  I want to accept it, let me do it。Do I need to close this pr first?

Reply:
> @SbloodyS I want to accept it, let me do it。Do I need to close this pr first?

Yes. Please reply in the issue #11902 so that I can assign to you.